common/djangoapps/student/views.py

```diff
@@ -2717,3 +2717,23 @@ def get_context_data(self, **kwargs):
         })
 
         return context
+
+
+class LogoutRedirectView(TemplateView):
+    """
+    Logs out user, performing the same logic as LogoutView, and redirects the user to a specified target.
+    """
+
+    template_name = 'logout.html'
+
+    def get_context_data(self, **kwargs):
+        context = super(LogoutRedirectView, self).get_context_data(**kwargs)
+        target = kwargs.get('redirect_to', '/')
+        logout_uri = "{}?{}".format(reverse('logout'), 'no_redirect=1')
+
+        context.update({
+            'target': target,
+            'logout_uri': logout_uri,
```

Comment:
Yes... okay. Rather than create a whole new view, apply a diff something like this to the existing logout view:

```diff
-    target = reverse_lazy('cas-logout') if settings.FEATURES.get('AUTH_USE_CAS') else '/'
+    default_target = reverse_lazy('cas-logout') if settings.FEATURES.get('AUTH_USE_CAS') else '/'
+
+    @property
+    def target(self):
+        target_url = self.request.GET.get('redirect_url')
+        return target_url or self.default_target
```

That seems to work for me.